pfcpiface/pfcpiface.go

```diff
@@ -372,3 +425,89 @@ func handleSessionDeletionRequest(upf *upf, msg message.Message, addr net.Addr,
 
 	return smres
 }
+
+func manageSmfConnection(n4LocalIP string, n3ip string, n4Dst string, conn *net.UDPConn, cpConnectionStatus chan bool) {
+	msg := false
+	cpConnected := false
+
+	initiatePfcpConnection := func() {
+		log.Println("SPGWC/SMF hostname ", n4Dst)
+		n4DstIP := getRemoteIP(n4Dst)
+		log.Println("SPGWC/SMF address IP inside manageSmfConnection ", n4DstIP.String())
+		// initiate request if we have control plane address available
+		if n4DstIP.String() != "0.0.0.0" {
+			generateAssociationRequest(n4LocalIP, n3ip, n4DstIP.String(), conn)
+		}
+		// no worry. Looks like control plane is still not up
+	}
+	updateSmfStatus := func(msg bool) {
+		log.Println("cpConnected : ", cpConnected, "msg ", msg)
+		//events from main Loop
+		if cpConnected && !msg {
+			log.Println("CP disconnected ")
+			cpConnected = false
+		} else if !cpConnected && msg {
+			log.Println("CP Connected ")
+			cpConnected = true
+		} else {
+			log.Println("cpConnected ", cpConnected, "msg - ", msg)
+		}
+	}
+
+	if n4Dst != "" {
+		go initiatePfcpConnection()
```

Comment:
this can be just a function call instead of a go routine

Reply:
Fixed.